_posts/2020-05-14-beginners-config.md

```diff
@@ -0,0 +1,116 @@
+
+*   Install and start up the dask processes [manually from the CLI](https://docs.dask.org/en/latest/setup/cli.html) on cloud instances you provision, such as AWS EC2 or GCP GCE
+*   Use popular deployment interfaces such as [helm for kubernetes](https://docs.dask.org/en/latest/setup/kubernetes-helm.html) to deploy dask in cloud container clusters you provision, such as AWS Fargate or GCP GKE
+*   Use ‘native’ deployment python APIs, provided by the dask developers, to create (and interactively configure) dask on deployment infrastructure they support, namely kubernetes with [dask-kubernetes](https://kubernetes.dask.org/en/latest/) and YARN with [dask-yarn](https://yarn.dask.org/en/latest/), as long as you’ve already provisioned the kubernetes cluster or hadoop cluster already
+*   Use a nearly full-service deployment python API called [Dask Cloud Provider](https://cloudprovider.dask.org/en/latest/), that will go one step farther and provision the cluster for you too, as long as you give it AWS credentials (and as of time of writing, it only supports AWS)
+
+As you can see, there are a ton of options. On top of all of those, you might contract a managed service provider to provision and configure your dask cluster for you according to your specs, such as Saturn Cloud.
+
+Whatever you choose, the whole point is to unlock the power of parallelism in Python that Dask provides, in as scalable a manner as possible which is what getting it running on distributed infrastructure is all about. Once you know where and with what API you are going to deploy your dask cluster, the real configuration process for your Dask cluster and its workload begins.
+
+
+## How to choose instance type for your cluster
+
+When you are ready to set up your dask cluster for production, you will need to make some decisions about the infrastructure your scheduler and your workers will be running on, especially if you are using one of the options from [How to host  a distributed dask cluster](#how-to-host-a-distributed-dask-cluster) that requires pre-provisioned infrastructure. Whether your infrastructure is on-prem or in the cloud, the classic decision points need to be made:
+
+*   Memory requirements
+*   CPU requirements
+*   Storage requirements
+
+If you have tested your flows locally, a simple heuristic is to multiply the CPU, storage, and memory usage of your flows by some multiplier that is related to how scaled down your local flow run is from your expected production usage. For example, if you test your flows locally with a 10% sample of data, multiplying any observed resource usage by at least 10 will get you close to your minimum instance size. Doing this for all flows that will be deployed, taking into account flow schedules that may cause flows to stagger in their concurrency, is a good first pass technique.
```

Comment:
"flows" -> "workflows"? Is "flow" more of a Prefect-specific thing?

Reply:
I guess some of this paragraph is prefect specific, but it'd be nice to make this guidance as generic as possible, followed up by specific examples / recommendations that might use prefect.

Reply:
Multiplying the size of the cluster by the size of the data may not make sense for workflows that can run in small space (which is most).  A big benefit of Dask is that you don't need 10x the memory to work on 10x the data.

Reply:
@TomAugspurger that was an oversight on my part, I had been writing some of it "prefect-y" at first and tried to tear all that out, but missed it here :) This blog is intended to have no prefect-specific examples/recommendations.

@mrocklin good point. That being said we were trying to go for some simple tips for people who are just jumping into optimization might try. I can soften the language for sure as for in terms of how much YMMV here